src/Raven.Server/Documents/ETL/EtlProcess.cs

```diff
@@ -422,14 +422,16 @@ public bool CanContinueBatch(EtlStatsScope stats, TExtracted currentItem, JsonOp
                 return false;
             }
 
-            var currentlyInUse = new Size(_threadAllocations.TotalAllocated, SizeUnit.Bytes);
+            var totalAllocated = _threadAllocations.TotalAllocated;
+            _threadAllocations.CurrentlyAllocatedForProcessing = totalAllocated;
+            var currentlyInUse = new Size(totalAllocated, SizeUnit.Bytes);
             if (currentlyInUse > _currentMaximumAllowedMemory)
             {
                 if (MemoryUsageGuard.TryIncreasingMemoryUsageForThread(_threadAllocations, ref _currentMaximumAllowedMemory,
                         currentlyInUse,
                         Database.DocumentsStorage.Environment.Options.RunningOn32Bits, Logger, out ProcessMemoryUsage memoryUsage) == false)
                 {
-                    var reason = $"Stopping the batch because cannot budget additional memory. Current budget: {new Size(_threadAllocations.TotalAllocated, SizeUnit.Bytes)}. Current memory usage: " +
+                    var reason = $"Stopping the batch because cannot budget additional memory. Current budget: {new Size(totalAllocated, SizeUnit.Bytes)}. Current memory usage: " +
```

Comment:
We can use 'currentlyInUse' here instead?

Reply:
done